File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i in range(0, len(nums)):
            dic[nums[i]] = i
        
        for i in range(0, len(nums)):
            if target - nums[i] in dic.keys() and dic[target - nums[i]] != i:
                return [i,dic[target - nums[i]]] 

File: 0001-two-sum/0001-two-sum.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val d_map = scala.collection.mutable.Map[Int, Int]()
        d_map += (nums(0)->0)
        var result: Array[Int] = Array()

        for(i <- 1 until nums.length)
        {   var comp = target - nums(i)
            if (d_map.contains(comp)){
                result = Array(d_map(comp),i)
            }
            d_map += (nums(i)->i)
        }
        return result
    }
}